### PR TITLE
2117 cache variables

### LINKED ISSUE #2117: Easier Customization of Desktop Client
### Expected behaviour
I guess there are often requests to make customization of the Nextcloud desktop client easier, additionaly to the Theme containing the Logo of the Application. This can be done, but it is realy cumbersome, because you need to modify the `NEXTCLOUD.cmake` File.

- Custom Application Name:  [Link 1](https://help.nextcloud.com/t/slight-customization/5000),  [Link 2](https://help.nextcloud.com/t/how-to-replace-the-name-nextcloud-with-my-name-in-the-desktop-client/21960)
- Custom Theme: [Link 1](https://help.nextcloud.com/t/branding-desktop-clients/5876/3), [Link 2](https://github.com/nextcloud/client_theming)

I guess it would be nice if you could make the .cmake-variables for the application modifiable as well.

```
    cmake "-G${VERSION_VSSTUDIO}" \
        "../build" \
        -Wno-dev \
        -D CMAKE_PREFIX_PATH=${DIR_QT_MSVC} \
        -D QTKEYCHAIN_LIBRARY="${DIR_QTKEYCHAIN}/lib/qt5keychain.lib" \
        -D QTKEYCHAIN_INCLUDE_DIR="${DIR_QTKEYCHAIN}/include/qt5keychain" \
        -D ZLIB_LIBRARY="${DIR_ZLIB}/lib/zlib.lib" \
        -D ZLIB_INCLUDE_DIR="${DIR_ZLIB}/include" \
        -D OPENSSL_ROOT_DIR="${DIR_OPENSSL}" \
        -D OPENSSL_INCLUDE_DIR="${DIR_OPENSSL}/include/" \
        -D OPENSSL_LIBRARIES="${DIR_OPENSSL}/lib/" \
        -D NO_SHIBBOLETH=1 \
        \
        -D APPLICATION_NAME="My Application" \
        -D APPLICATION_SHORTNAME="myapp" \
        \
        .
```

**As well as all other variables:** APPLICATION_EXECUTABLE, APPLICATION_DOMAIN, APPLICATION_VENDOR, APPLICATION_ICON_NAME, LINUX_PACKAGE_SHORTNAME, THEME_CLASS, APPLICATION_REV_DOMAIN, WIN_SETUP_BITMAP_PATH, MAC_INSTALLER_BACKGROUND_FILE, THEME_INCLUDE, APPLICATION_LICENSE, ...

With this approach it would be easier to change the application name, without patching your cmake file like approached in the links mentioned.

### PR DESCRIPTION
**Pull request by AHeimberger**
This branch allows users to cache NEXTCLOUD.cmake variables. This way the `APPLICATION_NAME `and further variables can be controlled by the user during the build configuration by using -D `APPLICATION_NAME="Application"`.

Fix #2117

### CODE REVIEW

**Comment by er-vin** [comment]
Overall looks good to me.

That said, could you please satisfy the DCO bot? Commits are expected to be signed off.

**Comment by sbeyer** [comment]
I wonder if many of these cache variables should be marked as advanced (using [`mark_as_advanced()`](https://cmake.org/cmake/help/v3.2/command/mark_as_advanced.html)) because it is quite unlikely for the average user to, e.g., rename the application or change the icon.

**Comment by er-vin** [comment]
> I wonder if many of these cache variables should be marked as advanced (using [`mark_as_advanced()`](https://cmake.org/cmake/help/v3.2/command/mark_as_advanced.html)) because it is quite unlikely for the average user to, e.g., rename the application or change the icon.

That's a good point indeed. Could you please do that as well as pleasing the DCO bot?

Thanks in advance.

**Comment by AHeimberger** [comment]
In my humble opinion those variables exist, cause someone wanted to change them application widely and it is a strange behavior when someone has to copy a file to modify them. 

However, do you have a specific wish? I could mark all variables which have not been modifiable before as advanced. There are still some variables I kept untouched like CRASHREPORTER_SUBMIT_URL, CRASHREPORTER_ICON or THEME_INCLUDE because they are commented.

**Comment by er-vin** [comment]
Looks good to me, just got the DCO bot to please and this can go in I think.

**Comment by er-vin** [comment]
> I approve the changes. However, I don't like the CMake coding style used in this file (but that's (a) a matter of taste, and (b) not your fault); especially not the aligning, and not the spaces after `(` and before `)`.

Well yeah... one thing at a time. ;-)

Most of the CMake stuff in the repository needs a serious clean up. As soon as DCO bot is satisfied I'll get this one rebased and merged.

**Comment by sbeyer** [comment]
> Most of the CMake stuff in the repository needs a serious clean up. As soon as DCO bot is satisfied I'll get this one rebased and merged.

I once had a "fix coding style" commit in my cmake-tidy branch, but I dropped it before I created the PR here ;-)

File: NEXTCLOUD.cmake [hunk 1]
```diff
@@ -1,37 +1,44 @@
-set( APPLICATION_NAME       "Nextcloud" )
-set( APPLICATION_SHORTNAME  "Nextcloud" )
-set( APPLICATION_EXECUTABLE "nextcloud" )
-set( APPLICATION_DOMAIN     "nextcloud.com" )
-set( APPLICATION_VENDOR     "Nextcloud GmbH" )
+mark_as_advanced( FORCE APPLICATION_NAME APPLICATION_SHORTNAME APPLICATION_EXECUTABLE APPLICATION_DOMAIN APPLICATION_VENDOR APPLICATION_ICON_NAME APPLICATION_REV_DOMAIN LINUX_PACKAGE_SHORTNAME THEME_CLASS )
```
Comment (FlexW) [inline NEXTCLOUD.cmake:1]: I think FORCE is not necessary. Also I would find it more convenient to place that `mark_as_advanced` statement at the end of the file.

**Comment by FlexW** [comment]
Are you still interested in that pr? @tobiasKaminsky @er-vin I think this pr could have a serious regression in brander or should it be fine?

**Comment by tobiasKaminsky** [comment]
> regression in brander

If you think this PR helps you in general, then we can adjust brander afterwards.

**Review by mgallien — APPROVED** [review]
thanks

File: NEXTCLOUD.cmake (hunk 1 above)
Comment (mgallien) [reply NEXTCLOUD.cmake:1]: even if you are probably true, setting `FORCE` has the advantage to ensure we get the desired state whatever is in the cmake cache

**Comment by Rello** [comment]
Hello,

Thank you for your contribution to the Desktop Client with this pull request.
We are closing this request as it is outdated, no longer relevant (e.g., due to Qt 6), or does not align with the current roadmap.

We truly value your support and hope to see more contributions from you in the future!


Best regards,
Nextcloud Desktop Team

**Pull request closed without merging**